aerogear/mobile-security-service-operator
# feat(OpenAPI) : Add schema validation and OpenAPI

## Motivation
https://issues.jboss.org/browse/AEROGEAR-9507

## What
- Add OpenAPI Schema and definitions

## Why
- It will make easier create the OLM
- Following the good practices
- Make easier the understatement
- Validate the data in the CR by the API

## Verification Steps
* Check the CI
* Run the command `make setup` which will call the make code/gen which updates the files automatically.  

## Checklist:


## Progress


 ## Extra

* if you wish you can test the operator again besides not be required since no impacts it should have. The image is: `docker.io/cmacedo/mobile-security-service-operator:openAPI`

* If you wish you can check how this validation works as follows. 

```yaml
apiVersion: mobile-security-service.aerogear.com/v1alpha1
kind: MobileSecurityService
metadata:
  name: mobile-security-service
spec:
   routeName: 1 # It will not be applied because there is expected a string
```
